### PR TITLE
Set null-count in linked_column_view conversion operator

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Removes the `UNKNOWN_NULL_COUNT` usage in the `linked_column_view::column_view()` conversion operator. The null-count is copied from the parent instance. The `linked_column_view` class was reworked to move the C++ function definitions from the header file to a new .cpp file.

Contributes to: https://github.com/rapidsai/cudf/issues/11968

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/utilities/linked_column.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+linked_column_view::operator column_view() const
+{
+  auto child_it = thrust::make_transform_iterator(
+    children.begin(), [](auto const& c) { return static_cast<column_view>(*c); });
+  return column_view(this->type(),
+                     this->size(),
+                     this->head(),
+                     this->null_mask(),
+                     this->null_count(),
```
Comment (vyasr) [inline cpp/src/utilities/linked_column.cpp:43]: This is the only change I see aside from the move (and it looks correct). Please re-request review if there is any other substantive change that I missed.

**Review by vyasr — APPROVED** [review]

File: cpp/include/cudf/detail/utilities/linked_column.hpp [hunk 2]
```diff
@@ -35,30 +37,11 @@ struct linked_column_view : public column_view_base {
   linked_column_view(linked_column_view const&) = delete;
   linked_column_view& operator=(linked_column_view const&) = delete;
 
-  linked_column_view(column_view const& col) : linked_column_view(nullptr, col) {}
+  linked_column_view(column_view const& col);
```
Comment (ttnghia) [inline cpp/include/cudf/detail/utilities/linked_column.hpp:40]: Recommend to remove the empty lines in this class, such as the ones right below this.

File: cpp/src/utilities/linked_column.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                     this->offset(),
+                     std::vector<column_view>(child_it, child_it + children.size()));
+}
+
+/**
+ * @brief Converts all column_views of a table into linked_column_views
+ *
+ * @param table table of columns to convert
+ * @return Vector of converted linked_column_views
+ */
```
Comment (ttnghia) [inline cpp/src/utilities/linked_column.cpp:53]: I believe that we just need to have doxygen in the header.

**Review by ttnghia — APPROVED** [review]

**Pull request merged**